lib/mongoose.js

```diff
@@ -0,0 +1,22 @@
+const m2s = require('mongoose-to-swagger'); 
+
+module.exports.generateMongooseModelsSpec = mongooseModelNames => {
+  try {
+    /* Disabling because its a peer dependency */
+    /* eslint-disable global-require */
+    /* @ts-ignore */
+    const mongoose = require('mongoose');
```

Comment:
I think you could require the `mongoose` package at the top-level scope, just like you've done with `m2s`. This will be more efficient since you won't have to re-require the `mongoose` package every time the function is called (it will instead only be required once - when the file loads).

Reply:
This is a good one 😄 It was intentional. We could include the `mongoose` package as a:

**a) Peer dependency**. 
This was the original idea. The installation is optional for the client (could be using Sequelize for example).
As we can't handle errors at top-level scope, it would throw in case we `require` an uninstalled package.
Even though it's less performant to `re-require` inside the `generateMongooseModelsSpec` scope, that function is intended to be called once and then use the same generated spec every time. 

Ignore that it's also included as a Dev dependency, that was for `basic_server.js` testing purposes.

**b) Regular dependency**. 
The installation would be required. It could lead to issues if the `mongoose package version` from the client is different from the one used as a dependency on `express-oas-generator`.
In this case, it would be correct to require at `top-level` scope.


For both cases, we should decide which is the minimum required version: https://mongoosejs.com/docs/compatibility.html

Reply:
> Shouldn't `const m2s = require('mongoose-to-swagger');` be moved inside `generateMongooseModelsSpec` as well to have it as peer dependency?

Sure, so should be go in both cases (`mongoose`, `mongoose-to-swagger`) with peer dependencies?

Reply:
They are all now peer dependencies (`bson, mongoose, mongoose-to-swagger`). We should now agree the minimum required versions.